db/db_impl.cc

```diff
@@ -907,6 +907,15 @@ InternalIterator* DBImpl::NewInternalIterator(
     internal_iter->RegisterCleanup(CleanupIteratorState, cleanup, nullptr);
 
     return internal_iter;
+  } else {
+    // cleanup super version.
```

Comment:
@ajkr I'm not familiar with the logic here. Should I call CleanupIteratorState instead?

Reply:
I think in other cases where we hold the SuperVersion for a very short time, like in `Get()`, we don't cleanup everything, so there's a small chance obsoleted files will still exist until the next background job happens.

You should use `ReturnAndCleanupSuperVersion`, though.

Reply:
The SuperVersion is obtained via `cfd->GetReferencedSuperVersion()` which is already returned to the ThreadLocalPtr. Here we only need the cleanup part, so `ReturnAndCleanupSuperVersion` does not work.